bftengine/src/bftengine/ReplicaConfigSerializer.cpp

```diff
@@ -117,14 +117,12 @@ void ReplicaConfigSerializer::serializeDataMembers(ostream &outStream) const {
   // Serialize replicaPrivateKey
   serializeKey(config_->replicaPrivateKey, outStream);
 
-  serializePointer(config_->thresholdSignerForSlowPathCommit, outStream);
-  serializePointer(config_->thresholdVerifierForSlowPathCommit, outStream);
-
-  serializePointer(config_->thresholdSignerForCommit, outStream);
-  serializePointer(config_->thresholdVerifierForCommit, outStream);
-
-  serializePointer(config_->thresholdSignerForOptimisticCommit, outStream);
-  serializePointer(config_->thresholdVerifierForOptimisticCommit, outStream);
+  // Serialize threshold crypto system
+  concord::serialize::Serializable::serialize(outStream, config_->thresholdSystemType_);
+  concord::serialize::Serializable::serialize(outStream, config_->thresholdSystemSubType_);
+  serializeKey(config_->thresholdPrivateKey_, outStream);  // TODO [TK] serialize to a secret key store
+  serializeKey(config_->thresholdPublicKey_, outStream);
```

Comment:
the same question as above, it's for the bootstrap only keys, right?

Reply:
Right.
Eventually the private key will be sent to a secret store and public keys will "live" in ReservedPages